### ISSUE TITLE
App crashes after prompting camera and microphone permissions

### ISSUE THREAD

**Issue by oedotme** [Gruveo/sdk-examples-android-java#10]
Hello,

We've Gruveo SDK working on Web, Android and IOS, And working on a new App built with React Native whose users will interact with those three platforms.

By the use of that library [react-native-gruveo](https://github.com/senseobservationsystems/react-native-gruveo) it starts only while prompting user for camera and microphone permissions, then it crashes the app.

Looking forward to solve the problem, Thanks in advance.

**Comment by akkewood** [comment]
Hello. Could you please provide logs with crash (stack trace) please?

**Comment by oedotme** [comment]
Hey, [Here](https://gist.github.com/iixixv/ea603c6f575ba3b25ed002e5c220ffc4) are logs outputted by `adb logcat`

**Comment by akkewood** [comment]
What sdk version are you using?
Our latest version is 0.6.4.  (implementation 'org.bitbucket.gruveo:gruveo-sdk-android:0.6.4')
If your version isn't latest please update it and check whether this crash happens on latest sdk version.

**Comment by oedotme** [comment]
I'm using version 0.5.6 as that [library](https://github.com/senseobservationsystems/react-native-gruveo)

I've updated to version 0.5.8 and here are [logs](https://gist.github.com/iixixv/76a57b4cce209b2ef5e202bbf80407e2).
Noting that it doesn't stop app from working and crashes as on version 0.5.6, instead it launches another (gruveo) activity that fails, then switches back to app main activity.
Second time to launch after permissions being prompted previously it crashes immediately

After updating to version 0.6.4, compilation failes with two errors
- Error: GruveoModule is not abstract and does not override abstract method recordingFilename(String) in EventsListener
  > GruveoModule.java :21

  ```java
  public class GruveoModule extends ReactContextBaseJavaModule implements Gruveo.EventsListener {
  ```

- Error: method toggleRecording in class Companion cannot be applied to given types;
  Required: boolean,int
  Found: boolean
  Reason: actual and formal argument lists differ in length
  > GruveoModule.java :118

  ```java
  public void toggleRecording(boolean enable) {
    Gruveo.Companion.toggleRecording(enable);
  }
  ```

**Comment by akkewood** [comment]
1You should override in your GruveoModule file 2 methods
1) 'recordingFilename(String)'
2) 'toggleRecording(boolean,int)

**Comment by oedotme** [comment]
After fixing previous errors,

- Error: cannot access Unit
class file for kotlin.Unit not found
  > GruveoModule.java :77

  ```java
  public void authorize(String token) {
    Gruveo.Companion.authorize(token);
  }
  ```

**Comment by akkewood** [comment]
You should add kotlin support. We updated our example.
For this purpose you should do 2 steps:
1) in base gradle file add 3 lines:
        ext.kotlin_version = "1.3.31"
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
        classpath "org.jetbrains.kotlin:kotlin-android-extensions:$kotlin_version"
you can see it in our example, file: /sdk-examples-android-java/build.gradle
2)in app gradle file add 3 lines:
         apply plugin: 'kotlin-android'
         apply plugin: 'kotlin-android-extensions'
         implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
you can see it in our example file: /sdk-examples-android-java/app/build.gradle

Also please update gruveo sdk in your project up to version number 0.6.5

**Comment by oedotme** [comment]
Issue has been fixed updating to 0.6.5
Working properly on demo id and signer
